src/main/java/com/teragrep/pth_06/planner/bloomfilter/SearchTermBloomFilter.java

```diff
@@ -0,0 +1,125 @@
+ *
+ * Names of the licensors and authors may not be used for publicity purposes.
+ *
+ * No rights are granted for use of trade names, trademarks, or service marks
+ * which are in The Program if any.
+ *
+ * Licensee must indemnify licensors and authors for any liability that these
+ * contractual assumptions impose on licensors and authors.
+ *
+ * To the extent this program is licensed as part of the Commercial versions of
+ * Teragrep, the applicable Commercial License may apply to this file if you as
+ * a licensee so wish it.
+ */
+package com.teragrep.pth_06.planner.bloomfilter;
+
+import org.apache.spark.util.sketch.BloomFilter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Inserts given tokens into configurable filter
+ */
+public final class SearchTermBloomFilter {
+
+    private final Logger LOGGER = LoggerFactory.getLogger(SearchTermBloomFilter.class);
+    private final Long expected;
+    private final Double fpp;
+    private final List<String> stringTokens;
+
+    public SearchTermBloomFilter(Long expected, Double fpp, RegexExtractedValue tokenizable) {
+        this(expected, fpp, tokenizable.tokens());
+    }
+
+    public SearchTermBloomFilter(Long expected, Double fpp, TokenizedValue tokenizable) {
+        this(expected, fpp, new TokensAsStrings(tokenizable).tokens());
+    }
+
+    public SearchTermBloomFilter(Long expected, Double fpp, List<String> stringTokens) {
+        this.expected = expected;
+        this.fpp = fpp;
+        this.stringTokens = stringTokens;
+    }
+
+    public byte[] bytes() {
+        LOGGER.debug("Create filter from Record with values: expected <{}>, fpp <{}>", expected, fpp);
+        if (stringTokens.isEmpty()) {
+            throw new IllegalStateException(
+                    "Trying to insert empty filter, pattern match joined table should always have tokens"
+            );
+        }
+        final BloomFilter filter = BloomFilter.create(expected, fpp);
+        for (final String token : stringTokens) {
+            filter.put(token);
+        }
+        if (LOGGER.isWarnEnabled()) {
+            if (stringTokens.size() > expected) {
+                LOGGER
+                        .warn(
+                                "Number of tokens <{}> was larger than the expected value <{}>, resulting FPP <{}>",
+                                stringTokens.size(), expected, filter.expectedFpp()
+                        );
+            }
+        }
+        try (final ByteArrayOutputStream filterBAOS = new ByteArrayOutputStream()) {
+            filter.writeTo(filterBAOS);
+            return filterBAOS.toByteArray();
+        }
+        catch (IOException e) {
+            throw new UncheckedIOException(new IOException("Error writing filter bytes: " + e.getMessage()));
```

Comment:
Instead of creating a new IOException here, the existing one could be passed to the UncheckedIOException using the other constructor: UncheckedIOException(String message, IOException cause)

Reply:
refactored to use constructor